editor.planx.uk/src/@planx/components/Pay/Public/Pay.tsx

```diff
@@ -133,40 +136,55 @@ function Component(props: Props) {
     props.handleSubmit(makeData(props, govUkPayment, GOV_PAY_PASSPORT_KEY));
   };
 
-  const updatePayment = (responseData: any): GovUKPayment => {
-    const payment: GovUKPayment = responseData;
-
-    const normalizedPayment = {
-      ...payment,
-      amount: toDecimal(payment.amount),
-    };
-
-    setGovUkPayment(normalizedPayment);
+  const normalizePaymentResponse = (responseData: any): GovUKPayment => {
+    if (!responseData?.state?.status)
+      throw new Error("Corrupted response from GOV.UK");
+    let payment: GovUKPayment = responseData;
+    if (payment.amount)
+      payment = {
+        ...payment,
+        amount: toDecimal(payment.amount),
+      };
+    return payment;
+  };
 
-    return normalizedPayment;
+  const resolvePaymentResponse = async (
+    responseData: any
+  ): Promise<GovUKPayment> => {
+    const payment = normalizePaymentResponse(responseData);
+    setGovUkPayment(payment);
+    try {
+      await createPaymentStatus({
+        teamSlug,
+        paymentId: payment.payment_id,
+        status: PaymentStatus[payment.state.status],
+      });
+    } catch (e) {
+      reportError({ error: e, payment });
+    }
+    return payment;
```

Comment:
what are your thoughts on benefits & trade-offs of keeping this request to update the audit table on the client-side versus moving into the two existing API endpoints for Pay using something like `responseInterceptor` from `https-proxy-middleware`? (find an example of this in send/bops)

my thinking from afar is: 
- "when" we want to update our audit table always aligns with when we hit our /pay API endpoints (i'm not sure there are any client-side only events that would trigger it?), so calling the `createPaymentStatus` mutation from the API-only seems fitting?
- the call to update this audit table is loads lighter and quicker than the original call to `session_backups`, but we know that the source of the previous unable-to-connect-to-gov-pay bug was a still-running client request when the user tried to connect, so maybe moving it out of the client entirely is better piece of mind going forward?

Reply:
Direct link from dev call (this is also a POST that intercepts the response and creates an audit table entry referencing the response data): https://github.com/theopensystemslab/planx-new/blob/main/api.planx.uk/send/bops.js#L65 (& sorry I wasn't specific enough about this example in my original comment)

If that's no help though or the trade-offs feel too significant, then happy for this to stay on the frontend for right now